### DISCUSSION TITLE
Send request using NTLM Authorization

### DISCUSSION THREAD

**Discussion by carlo-andreoli** [sindresorhus/got#2106] [Q&A] [answered]
Hi,

is there a way to use the package to send request using NTLM Authentication (username & password)

Thanks in advance.

**Comment by szmarczak** [comment]
Can you provide me with a link to the specification?

**Reply by carlo-andreoli** [reply]
I don't have any specification, simply what I was trying to achive is to execute this request made on postman with NTLM authorization:

<img width="1251" alt="Screenshot 2022-08-12 at 09 59 33" src="https://user-images.githubusercontent.com/106250309/184311157-aa3d1838-dd74-4cc4-aed5-a707dad52156.png">

**Comment by sindresorhus** [comment, marked as answer]
Got doesn't have built-in NTLM support. NTLM requires a multi-step challenge-response authentication process that doesn't fit well with Got's architecture.

Your best options are.

Use a dedicatd NTLM library like `@node-ntlm/httpreq`:

```javascript
import {ntlm} from '@node-ntlm/httpreq';

const response = await ntlm({
	url: 'https://example.com/api',
	username: 'username',
	password: 'password',
	domain: 'DOMAIN' // Optional
});

console.log(response.body);
```

Use a local NTLM proxy:

Set up a proxy like [CNTLM](http://cntlm.sourceforge.net/) or [Px](https://github.com/genotrance/px) to handle NTLM authentication locally, then use Got through the proxy:

```javascript
import got from 'got';
import {HttpsProxyAgent} from 'hpagent';

await got('https://example.com/api', {
	agent: {
		https: new HttpsProxyAgent({
			proxy: 'http://localhost:3128' // Your NTLM proxy
		})
	}
});
```

The dedicated NTLM libraries are your best bet since they handle the complex authentication handshake that NTLM requiers.
